### PR TITLE
Fix writebuffers and make sure service tests work again. 

### PR DESCRIPTION
**Pull request by frznlogic**

### CODE REVIEW

File: common/filetoolkitwithundo.cpp [hunk 1]
```diff
@@ -111,16 +111,20 @@ bool FileToolkitWithUndo::bindMount(const std::string &src,
                                     bool readOnly,
                                     bool enableWriteBuffer)
 {
-    unsigned long flags = MS_BIND;
+    unsigned long flags =  0; //MS_BIND;
```
Comment (DunderRoffe) [inline common/filetoolkitwithundo.cpp]: Is this a miss or are there a reason to keep the comment?

File: common/filetoolkitwithundo.cpp [hunk 2]
```diff
@@ -111,16 +111,20 @@ bool FileToolkitWithUndo::bindMount(const std::string &src,
...
 
     if (!existsInFileSystem(src)) {
         log_error() << src << " does not exist on the host, can not bindMount";
         return false;
     }
 
+    if (!existsInFileSystem(dst)) {
+        log_error() << src << " does not exist on the host, can not bindMount";
+        return false;
+    }
```
Comment (DunderRoffe) [inline common/filetoolkitwithundo.cpp]: code duplication

File: common/filetoolkitwithundo.cpp [hunk 3]
```diff
@@ -170,15 +178,15 @@ bool FileToolkitWithUndo::overlayMount(const std::string &lower,
                                        const std::string &dst)
 {
     std::string fstype = "overlay";
-    unsigned long flags = MS_BIND;
+    unsigned long flags = 0; //MS_BIND;
```
Comment (DunderRoffe) [inline common/filetoolkitwithundo.cpp]: Is this a miss or are there a reason to keep the comment?

File: common/recursivedelete.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+#include <ftw.h>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include <glibmm.h>
+
+namespace softwarecontainer {
+
+std::string deleteRoot;
```
Comment (DunderRoffe) [inline common/recursivedelete.cpp]: There should be a better solution than using global variables

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

File: common/filetoolkitwithundo.cpp (hunk 1 above)
Comment (frznlogic) [reply common/filetoolkitwithundo.cpp]: Both. I'm removing it.

File: common/filetoolkitwithundo.cpp (hunk 3 above)
Comment (frznlogic) [reply common/filetoolkitwithundo.cpp]: Both. I'm removing it ;).

File: common/filetoolkitwithundo.cpp (hunk 2 above)
Comment (frznlogic) [reply common/filetoolkitwithundo.cpp]: Actually no... code copy, but checking for src in the first test, then dst, but forgot to replace src with dst in the log message. I've fixed that. I hope that is good enough?

File: common/recursivedelete.cpp (hunk 4 above)
Comment (frznlogic) [reply common/recursivedelete.cpp]: Actually it's dead. Removed it.

File: common/recursivedelete.h [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    bool del(std::string dir);
+
+private:
+    RecursiveDelete();
+    ~RecursiveDelete();
+
+    /**
+     * @brief m_deleteLock lock used to protect global variables in the delete() method.
+     */
+    std::mutex m_deleteLock;
```
Comment (DunderRoffe) [inline common/recursivedelete.h]: I guess this should be removed as well now that the global variables are removed.

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

**Review by DunderRoffe — APPROVED** [review]

File: common/recursivedelete.h (hunk 5 above)
Comment (frznlogic) [reply common/recursivedelete.h]: Done!

**Review by breakreturn — APPROVED** [review]
Looks good, didn't comment on what Viktor said about duplication, but approving assuming that will be fixed.

**Pull request merged**